From bc34c37ed8265250c4c41c18953ab82fcc491c83 Mon Sep 17 00:00:00 2001
From: Hongyang Zhao <hongyang.zhao@thundersoft.com>
Date: Thu, 13 Aug 2026 19:53:34 +0800
Subject: [PATCH 13762/15333] ASoC: dt-bindings: es8316: Fix supply property
 constraints

The DT meta-schema requires a `then` clause when an `if` condition has
an `else` clause. Invert the compatible check and move the supply
property restrictions to `then` so they remain allowed only for ES8316.

Fixes: e9966d450b46 ("ASoC: dt-bindings: es8316: Add regulator supplies")
Reported-by: Rob Herring <robh@kernel.org>
Closes: https://lore.kernel.org/r/20260812194234.GA693895-robh@kernel.org
Signed-off-by: Hongyang Zhao <hongyang.zhao@thundersoft.com>
Link: https://patch.msgid.link/20260813-b4-es8316-binding-conditional-fix-v1-1-6cd56aa1370c@thundersoft.com
Signed-off-by: Mark Brown <broonie@kernel.org>
---
 .../devicetree/bindings/sound/everest,es8316.yaml          | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/Documentation/devicetree/bindings/sound/everest,es8316.yaml b/Documentation/devicetree/bindings/sound/everest,es8316.yaml
index 276c73bb4790..f4ff23120c5b 100644
--- a/Documentation/devicetree/bindings/sound/everest,es8316.yaml
+++ b/Documentation/devicetree/bindings/sound/everest,es8316.yaml
@@ -33,9 +33,10 @@ allOf:
   - if:
       properties:
         compatible:
-          contains:
-            const: everest,es8316
-    else:
+          not:
+            contains:
+              const: everest,es8316
+    then:
       properties:
         avdd-supply: false
         cpvdd-supply: false
-- 
2.55.0

